src/main/java/com/hubspot/smtp/messages/CrlfTerminatingChunkedStream.java

```diff
@@ -0,0 +1,38 @@
+package com.hubspot.smtp.messages;
+
+import java.io.InputStream;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.handler.stream.ChunkedStream;
+
+class CrlfTerminatingChunkedStream extends ChunkedStream {
+  private static final byte CR = '\r';
+  private static final byte LF = '\n';
+  private static final int DEFAULT_CHUNK_SIZE = 8192;
+  private static final byte[] TRAILING_BYTES = { CR, LF };
+
+  CrlfTerminatingChunkedStream(InputStream in) {
+    this(in, DEFAULT_CHUNK_SIZE);
+  }
+
+  CrlfTerminatingChunkedStream(InputStream in, int chunkSize) {
+    super(in, chunkSize);
+  }
+
+  @Override
+  public ByteBuf readChunk(ByteBufAllocator allocator) throws Exception {
+    ByteBuf chunk = super.readChunk(allocator);
+
+    int length = chunk.readableBytes();
+    if (!isEndOfInput()) {
+      return chunk;
+    }
+
+    if (length >= 2 && chunk.getByte(length - 2) == CR && chunk.getByte(length - 1) == LF) {
+      return chunk;
+    }
+
+    return allocator.compositeBuffer(2).addComponents(true, chunk, allocator.buffer(2).writeBytes(TRAILING_BYTES));
```

Comment:
in this mode, who is responsible for the final `\r\n.`? Is that part of dot stuffing?